agent/agent-profiler/agent-diagnostics-jfr/src/main/java/com/microsoft/applicationinsights/diagnostics/jfr/CGroupData.java

```diff
@@ -20,63 +23,110 @@
 @Description("CGroupData")
 @StackTrace(false)
 @Period("beginChunk")
-public class CGroupData extends Event {
+public class CGroupData extends Event implements JsonSerializable<CGroupData> {
 
   public static final String NAME = "com.microsoft.applicationinsights.diagnostics.jfr.CGroupData";
   public static final int CGROUP_DATA_ABSENT = -2;
 
   // Limit of the kernel memory
-  private final long kmemLimit; // /sys/fs/cgroup/memory/memory.kmem.limit_in_bytes
+  private long kmemLimit; // /sys/fs/cgroup/memory/memory.kmem.limit_in_bytes
 
   // Limit of the containers memory
-  private final long memoryLimit; // /sys/fs/cgroup/memory/memory.limit_in_bytes
+  private long memoryLimit; // /sys/fs/cgroup/memory/memory.limit_in_bytes
 
   // Soft memory limit (enforced over the long term)
-  private final long memorySoftLimit; // /sys/fs/cgroup/memory/memory.soft_limit_in_bytes
+  private long memorySoftLimit; // /sys/fs/cgroup/memory/memory.soft_limit_in_bytes
 
   // CPU usage limit
-  private final long cpuLimit; // /sys/fs/cgroup/cpu,cpuacct/cpu.cfs_quota_us
+  private long cpuLimit; // /sys/fs/cgroup/cpu,cpuacct/cpu.cfs_quota_us
 
   // CPU usage period
-  private final long cpuPeriod; // /sys/fs/cgroup/cpu,cpuacct/cpu.cfs_quota_us
-
-  @JsonCreator
-  public CGroupData(
-      @JsonProperty(value = "kmemLimit") long kmemLimit,
-      @JsonProperty(value = "memoryLimit") long memoryLimit,
-      @JsonProperty(value = "memorySoftLimit") long memorySoftLimit,
-      @JsonProperty(value = "cpuLimit") long cpuLimit,
-      @JsonProperty(value = "cpuPeriod", required = false) Long cpuPeriod) {
-    this.kmemLimit = kmemLimit;
-    this.memoryLimit = memoryLimit;
-    this.memorySoftLimit = memorySoftLimit;
-    this.cpuLimit = cpuLimit;
-
-    if (cpuPeriod == null) {
-      // No CGroup data was found for this value
-      this.cpuPeriod = CGROUP_DATA_ABSENT;
-    } else {
-      this.cpuPeriod = cpuPeriod;
-    }
-  }
+  private long cpuPeriod = CGROUP_DATA_ABSENT; // /sys/fs/cgroup/cpu,cpuacct/cpu.cfs_quota_us
```

Comment:
```suggestion
  private long cpuPeriod = CGROUP_DATA_ABSENT; // No CGroup data was found for this value
```

Reply:
this is old code. it can be a different PR and need @johnoliver input.

Reply:
ah, I see what happened, two lines got merged together and each one had a comment so one of the comments got lost

maybe move the comment that got lost up to the `CGROUP_DATA_ABSENT` constant declaration?

Reply:
which comment got lost? i didn't touch comments.

Reply:
this is the comment that got lost:

> // No CGroup data was found for this value